pkg/roachprod/vm/gce/gcloud.go

```diff
@@ -285,8 +289,8 @@ func (jsonVM *jsonVM) toVM(
 		RemoteUser:             remoteUser,
 		VPC:                    vpc,
 		MachineType:            machineType,
-		CPUArch:                vm.ParseArch(cpuPlatform),
```

Comment:
Similarly to the AWS case, can't we assume we already know `cpuPlatform`? Also note a couple of other side-effects of the change; `cpuPlatform` is still used below to derive `CPUFamily`; if any call to `GetVMArchitecture` fails for any reason, `List` fails.

Reply:
The CPU platform returned by `gcloud compute instances list` is something like `Intel Ice Lake`, which is not what we want here (especially with `fips` being considered as a CPU arch).

That being said, if we go the route of detecting `fips` from the AMI used, it means that I will force the `opts.includeVolumes` behavior and run `gcloud compute disks list` for all VMs, which returns the CPU architecture.